src/mlpack/methods/logistic_regression/logistic_regression.hpp

```diff
@@ -53,6 +53,30 @@ class LogisticRegression
   LogisticRegression(const MatType& predictors,
                      const arma::Row<size_t>& responses,
                      const double lambda = 0);
+  /**
+   * Construct the LogisticRegression class with the given labeled training
+   * data.  This will train the model.  Optionally, specify lambda, which is the
+   * penalty parameter for L2-regularization.  If not specified, it is set to 0,
+   * which results in standard (unregularized) logistic regression.
+   *
+   * It is not possible to set a custom optimizer with this constructor.  Either
+   * use a constructor that does not train and call Train() with a custom
+   * optimizer type, or use the constructor that takes an instantiated
+   * optimizer.  (This unfortunate situation is a language restriction of C++.)
+   *
+   * This constructor is necessary for the cross-validation code and
+   * hyper-parameter tuner to work, but it isn't suggested for regular use since
+   * it requires the extra numClasses parameter (which must always be 2).
+   *
+   * @param predictors Input training variables.
+   * @param responses Outputs resulting from input training variables.
+   * @param numClasses Number of classes in the dataset (must be 2).
+   * @param lambda L2-regularization parameter.
+   */
+  LogisticRegression(const MatType& predictors,
+                     const arma::Row<size_t>& responses,
+                     const size_t numClasses,
```

Comment:
Sorry, I missed this reply. The main disadvantage of keeping `numClasses` in `LogisticRegression` is that users will be required to pass it when they use `LogisticRegression` with the cross-validation and hyper-parameter tuning modules. If we want to keep it for the reason described in the documentation "as necessary for the cross-validation code and hyper-parameter tuner to work", it is not actually the case - cross-validation and hyper-parameter tuning will work without it. So, we need to keep it only if there is some other reason(s). Is there any other code that depends on the assumption of the presence of `numClasses` in classifiers?